scrapy/utils/request.py

```diff
@@ -106,3 +107,26 @@ def referer_str(request: Request) -> Optional[str]:
     if referrer is None:
         return referrer
     return to_unicode(referrer, errors='replace')
+
+
+def request_from_dict(d: dict, spider: Optional[Spider] = None) -> Request:
+    """Create Request object from a dict.
+
+    If a spider is given, it will try to resolve the callbacks looking at the
+    spider for methods with the same name.
+    """
+    request_cls = load_object(d["_class"]) if "_class" in d else Request
+    kwargs = {key: value for key, value in d.items() if key in request_cls._attributes}
```

Comment:
Since we expect Request subclasses to extend that variable as needed, maybe it should not be private.

Reply:
Makes sense